composer/datasets/cifar.py

```diff
@@ -0,0 +1,168 @@
+        dataset_name (str): Key used to determine where dataset is cached on local filesystem.
+        n_train_samples (int): Number of training samples.
+        n_val_samples (int): Number of validation samples.
+        height (int): Sample image height in pixels.
+        width (int): Sample image width in pixels.
+        n_classes (int): Number of output classes.
+        channel_means (list of float): Channel means for normalization.
+        channel_stds (list of float): Channel stds for normalization.
+    """
+
+    def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHparams) -> DataLoader:
+        if self.use_synthetic:
+            dataset = SyntheticBatchPairDataset(
+                total_dataset_size=self.n_train_samples if self.is_train else self.n_val_samples,
+                data_shape=[3, self.height, self.width],
+                num_classes=self.n_classes,
+                num_unique_samples_to_create=self.synthetic_num_unique_samples,
+                device=self.synthetic_device,
+                memory_format=self.synthetic_memory_format,
+            )
+        else:
+            if self.is_train:
+                split = 'train'
+                transform = transforms.Compose([
+                    transforms.RandomCrop((self.height, self.width), (self.height // 8, self.width // 8)),
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize(self.channel_means, self.channel_stds),
+                ])
+            else:
+                split = 'val'
+                transform = transforms.Compose([
+                    transforms.ToTensor(),
+                    transforms.Normalize(self.channel_means, self.channel_stds),
+                ])
+            dataset, meta = load_webdataset(self.dataset_s3_bucket, self.dataset_name, split, self.webdataset_cache_dir,
+                                            self.webdataset_cache_verbose)
+            if self.shuffle:
+                dataset = dataset.shuffle(self.shuffle_buffer_per_worker)
+            dataset = dataset.decode('pil').map_dict(jpg=transform).to_tuple('jpg', 'cls')
+            dataset = size_webdataset(dataset, meta['n_shards'], meta['samples_per_shard'], dist.get_world_size(),
+                                      dataloader_hparams.num_workers, batch_size, self.drop_last)
+        return dataloader_hparams.initialize_object(dataset,
+                                                    batch_size=batch_size,
+                                                    sampler=None,
+                                                    drop_last=self.drop_last)
+
+
+@dataclass
+class CIFAR10WebDatasetHparams(CIFARWebDatasetHparams):
+    """Defines an instance of the CIFAR-10 WebDataset for image classification."""
+
+    dataset_s3_bucket = 'mosaicml-internal-dataset-cifar10'
+    dataset_name = 'cifar10'
```

Comment:
need to use `hp.required` and `hp.optonal` with `import yahp as hp` for these to work. See the other hprams objects for reference.

Reply:
Sorry, what do you do when they're meant to be constants?